### PR TITLE
ci: split the tracer suite up into ten smaller suites

### PR DESCRIPTION
**Pull request by emmettbutler**
This change attempts to resolve common `exit code -9` failures in the `tracer` test suite by splitting it into many smaller units. This is based on the hypothesis that the exit code indicates the process having been killed due to OOM.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by P403n1x87** [comment]
Could we have this split done automatically by the `scripts/gen_circleci_config.py` script instead? This would also avoid us having to manually update these suites when new sources are added.

**Comment by emmettbutler** [comment]
@P403n1x87 sure! separate from this change, I'll look into having the script automatically split up large suites.

**Review by brettlangdon — APPROVED** [review]
nice

**Comment by mabdinur** [comment]
We have quite a few tracer tests that generate spans but do not call span.finish. This is likely causing the memory issues described in the PR. This change should hopefully address this: https://github.com/DataDog/dd-trace-py/commit/c294c9b76aa1415c55f07a93b4c283cd5b7d03f6. 

I will get this change to a mergeable state ASAP

**Comment by mabdinur** [comment]
> @mabdinur do you think there's a downside to this change that makes it not worth merging?

No concrete downsides. I am just afraid this change will make the tracer test suite harder to maintain. It would be nice if we could provide this parallelism without updating riot or suitespec.

File: tests/tracer/test_trace_utils.py [hunk 1]
```diff
@@ -402,13 +402,16 @@ def test_set_http_meta(
...
     cookies,
-    appsec_enabled,
     span_type,
+    appsec_enabled,
 ):
     int_config.http.trace_headers(["my-header"])
     int_config.trace_query_string = True
     span.span_type = span_type
     with override_global_config({"_asm_enabled": appsec_enabled}):
+        from ddtrace.appsec import _handlers
```
Comment (ZStriker19) [inline tests/tracer/test_trace_utils.py:412]: Was this just missing before and causing failures?

**Comment by ZStriker19** [comment]
Not maybe a great reason, but a reason: The way this breaks up the tracer tests is kind of brutal with regards to rerunning them if they fail. There are now 90 tests for tracer so re-running if pieces fail sounds pretty cumbersome. Maybe we see if @mabdinur 's PR fixes the issue and if not, go this route?

**Pull request closed without merging**